underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -148,17 +153,23 @@ public OutputStream create(String path) throws IOException {
   @Override
   public OutputStream create(String path, CreateOptions options) throws IOException {
     LOG.debug("Create method: {}", path);
-    String newPath = path.substring(Constants.HEADER_SWIFT.length());
-    if (newPath.endsWith("_SUCCESS")) {
+
+    if (!mkdirs(getParentKey(path), true)) {
+      return null;
+    }
+
+    final String strippedPath = stripPrefixIfPresent(path, Constants.HEADER_SWIFT);
+    //TODO(adit): this hack should not be here
+    if (strippedPath.endsWith("_SUCCESS")) {
       // when path/_SUCCESS is created, there is need to create path as
       // an empty object. This is required by Spark in case Spark
       // accesses path directly, bypassing Alluxio
-      String plainName = newPath.substring(0, newPath.indexOf("_SUCCESS"));
+      String plainName = strippedPath.substring(0, strippedPath.indexOf("_SUCCESS"));
```

Comment:
I saw this patter is used a lot, not only in this file but also in other ufs, to strip a suffix if the suffix presents. does it make sense to have a util function in common and making all of them calling the util function, instead of maintaining this method in each individual ufs. Thoese methods are now only working for the predefined suffix, we may want it a bit more general for any given suffix